build.sh

```diff
@@ -8,26 +8,14 @@ if ! command -v $SUDO; then
 	SUDO=""
 fi
 
-if command -v apt-get; then
-	if dpkg -l python3-venv; then
-		echo "python3-venv is installed, skipping setup"
-	else
-		if ! apt info python3-venv; then
-			echo python3-venv package info not found, trying apt update
-			$SUDO apt-get -qq update
-		fi
-		$SUDO apt-get install -qqy python3-venv
-	fi
-else
-	echo Skipping tool installation because your platform is missing apt-get.
-	echo If you see failures below, install the equivalent of python3-venv for your system.
+if ! command -v uv 2>&1 >/dev/null; then
+    pip install uv
 fi
 
 source .env
-echo creating virtualenv at $VIRTUAL_ENV
-python3 -m venv $VIRTUAL_ENV
-echo installing dependencies from requirements.txt
-$VIRTUAL_ENV/bin/pip install -r requirements.txt
-source .venv/bin/activate 
-$PYTHON -m PyInstaller --onefile --hidden-import="googleapiclient" --add-data "./src/models/checkpoints:checkpoints"  src/main.py
+# the version of pytorch we are using requires 3.11 or lower
+uv venv --python 3.11
```

Comment:
Note: realized that builds break with python 3.12, so am using uv to lock this version.  This will work both with local builds and cloud CI builds